packages/experimental/src/inbox-note/action.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { ADMIN_URL as adminUrl } from '@woocommerce/wc-admin-settings';
+
+type InboxNoteActionProps = {
+	onClick: () => void;
+	label: string;
+	href?: string;
+};
+
+/**
+ * Renders a secondary button that can also be a link. If href is provided it will
+ * automatically open it in a new tab/window.
+ */
+const InboxNoteAction: React.FC< InboxNoteActionProps > = ( {
+	label,
+	onClick,
+	href,
+} ) => {
+	const [ inAction, setInAction ] = useState( false );
+
+	const handleActionClick: React.MouseEventHandler = ( event ) => {
+		const target: EventTarget = event.target;
+		const targetHref = target instanceof HTMLBaseElement ? target.href : '';
+		let isActionable = true;
+
+		if ( targetHref.length && ! targetHref.startsWith( adminUrl ) ) {
+			event.preventDefault();
+			isActionable = false; // link buttons shouldn't be "busy".
+			window.open( targetHref, '_blank' );
+		}
+
+		setInAction( isActionable );
+		onClick();
+	};
+
+	return (
+		<Button
+			isSecondary
+			isBusy={ inAction }
+			disabled={ inAction }
+			href={ href }
+			onClick={ handleActionClick }
+		>
+			{ label }
+		</Button>
+	);
+};
+
+export default InboxNoteAction;
```

Comment:
Can we please avoid exporting defaults, it makes mass refactor's much harder due if anyone mis-names an import